snapfaas/src/worker.rs

```diff
@@ -313,15 +317,16 @@ impl<B: BackingStore> Worker<B> {
     }
 
     fn try_allocate(&self, f: &Function, payload_label: &Buckle) -> Option<Vm> {
-        if let Some(vm) = self.localrm.lock().unwrap().get_cached_vm(f) {
+        let mut localrm = self.localrm.lock().unwrap();
+        if let Some(vm) = localrm.get_cached_vm(f) {
             // cached VM must NOT be too tainted
             if !vm.label.can_flow_to(payload_label) {
```

Comment:
I don't remember why I had the negator `!` in the first place but now I think we should remove the negator `!`.
We can only reuse the VMs whose labels are lower than or equal to, i.e. can flow to, that of the payload.

Reply:
Yes, that seems right and, indeed I had on my todo list to see why we never actually reuse VMs in practice.

Logically, we always need to taint the VM with the payload label for security. For "correctness" (for the function to actually do useful things) we want the VM's starting label to be <= the payload_label. So, in practice indeed achieving both is to check the latter first (`vm.label.can_flow_to(payload_label)`) then do the former.

Reply:
Fixed this. This, unsurprisingly, makes a huge difference in performance.